cpp/src/fil/infer.cu

```diff
@@ -96,14 +96,14 @@ __device__ __forceinline__ vec<NITEMS, output_type> infer_one_tree(
   tree_type tree, const float* input, int cols, int n_rows) {
   int curr[NITEMS];
   // the first n_rows are active
-  int mask = ((1 << n_rows) - 1) << (NITEMS - n_rows);
+  int mask = (1 << n_rows) - 1;
```

Comment:
fixing a bug which somehow did not trigger the test in the previous PR. It triggers it now.